lib/src/component_declaration/component_type_checking.dart

```diff
@@ -231,20 +232,27 @@ bool isComponentOfType(ReactElement instance, dynamic typeAlias, {
     return false;
   }
 
+  var finalInstance = instance;
+  var instanceType = finalInstance.type;
```

Comment:
#nit (may not be correct / necessary. If so, I'd be curious why var is better :) )
```suggestion
  dynamic instanceType = finalInstance.type;
```

Reply:
`var` is preferred for local variables where the type can be inferred from the assignment.

Everywhere else, typing is preferred.

https://dart.dev/guides/language/effective-dart/design#avoid-type-annotating-initialized-local-variables

Reply:
💯